vpd-manager/include/utility/dbus_utility.hpp

```diff
@@ -466,5 +466,70 @@ inline bool isBMCReady()
 
     return false;
 }
+
+/**
+ * @brief An API to enable BMC reboot guard
+ *
+ * This API does a D-Bus method call to enable BMC reboot guard.
+ *
+ * @return On success, returns 0, otherwise returns -1.
+ */
+inline int EnableRebootGuard() noexcept
+{
+    int l_rc{-1};
+    try
+    {
+        auto l_bus = sdbusplus::bus::new_default();
+        auto l_method = l_bus.new_method_call(
+            "org.freedesktop.systemd1", "/org/freedesktop/systemd1",
+            "org.freedesktop.systemd1.Manager", "StartUnit");
+        l_method.append("reboot-guard-enable.service", "replace");
+        l_bus.call_noreply(l_method);
+        l_rc = 0;
+    }
+    catch (const sdbusplus::exception::SdBusError& l_ex)
+    {
+        std::string l_errMsg =
+            "D-Bus call to enable BMC reboot guard failed for reason: ";
+        l_errMsg += l_ex.what();
+
+        logging::logMessage(l_errMsg);
```

Comment:
you can directly append the log  here ?

just return from here and from the caller API, we can have log along with vpd path, so will know for which FRU this is failed, we can have log or PEL there.

Reply:
Add informational PEL here.

Reply:
> you can directly append the log here ?

> just return from here and from the caller API, we can have log along with vpd path, so will know for which FRU this is failed, we can have log or PEL there.

As per discussion, we want to add Informational PEL here. Its better that the API body itself logs this PEL rather than leaving it up to the caller to log the Informational PEL.

> we can have log along with vpd path, so will know for which FRU this is failed.

Enable/Disable Reboot Guard failure is not related to any FRU.

Reply:
I would suggest return the error log from here, let the caller decide what needs to be done in case of failure and what kind of PEL should be logged.
This is a utility method can be used at multiple places if required. Having a common log here will make it difficult to differentiate which operation failed.

Reply:
Addressed this. The caller of the API should check for the return value and take necessary action.